examples/ecs/state.rs

```diff
@@ -6,26 +6,31 @@ use bevy::prelude::*;
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
-        .add_state(AppState::Menu)
+        .add_state::<AppState>()
         .add_startup_system(setup)
-        .add_system_set(SystemSet::on_enter(AppState::Menu).with_system(setup_menu))
-        .add_system_set(SystemSet::on_update(AppState::Menu).with_system(menu))
-        .add_system_set(SystemSet::on_exit(AppState::Menu).with_system(cleanup_menu))
-        .add_system_set(SystemSet::on_enter(AppState::InGame).with_system(setup_game))
-        .add_system_set(
-            SystemSet::on_update(AppState::InGame)
-                .with_system(movement)
-                .with_system(change_color),
-        )
+        .add_system_to_schedule(OnEnter(AppState::Menu), setup_menu)
+        .add_system(menu.on_update(AppState::Menu))
```

Comment:
This could use some comments on why on_update is different from OnExit, OnEnter.